### ISSUE TITLE
[BUG] Stored XSS

### ISSUE THREAD

**Issue by nk521** [rshrc/NSP#110]
**Stored XSS during profile picture upload**

**To Reproduce**
* Sail to [this](https://thensp.pythonanywhere.com/account/change_profilepic)
* Upload a file with Vuln-JS
* Done! Now If anyone opens my profile picture that JS will be executed

**Expected behavior**
I expected that there'll be certain check whether the file is an image type or not.. but nah!

**Screenshots**
Sorry but no SS

**Desktop (please complete the following information):**
* Arch running Firefox 61.0.1

**Additional context**
Worst case will be stealing cookies.. If you're just checking about the metadata and magicnumber stuff of a file which is being uploaded then look at this example : 

```
W equ 10799
H equ 69
db 'GIF89a'
    dw W, H
db 0
    db -1
    db 0
db 02ch
    dw 0, 0
    dw W, H
    db 0
db 2
db 0
db 3bh
db '*/'
db '=1;'
db '<script>alert("XSS")</script>;'
```

Compile it with `yasm -o mason.gif source.asm`.. resulting gif can be uploaded to server as an html source.. Your check will give a green signal to it as it satisfies each check (metadata and magicnumber) but still is a JS source..

A proper implementation of AJAX file upload will do the work for you

**Comment by gjergjk71** [comment]
Fixed with pull #113 
@nk521 thanks for reporting this bug, quite frankly I'm ashamed even though I didn't work on this part of the project.

**Comment by nk521** [comment]
Don't be ashamed dude.. it's how you learn not to write vuln code.. 

kudos!  @gjergjk71
